parity-crypto/src/publickey/ecdsa_signature.rs

```diff
@@ -0,0 +1,316 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+//! ECDSA signature
+
+use std::ops::{Deref, DerefMut};
+use std::cmp::PartialEq;
+use std::fmt;
+use std::str::FromStr;
+use std::hash::{Hash, Hasher};
+use secp256k1::{Message as SecpMessage, RecoverableSignature, RecoveryId, Error as SecpError};
+use secp256k1::key::{SecretKey, PublicKey};
+use rustc_hex::{ToHex, FromHex};
+use ethereum_types::{H520, H256};
+use super::{Secret, Public, SECP256K1, Message, public_to_address, Address, Error};
+
+/// Signature encoded as RSV components
+#[repr(C)]
+pub struct Signature([u8; 65]);
+
+impl Signature {
+	/// Get a slice into the 'r' portion of the data.
+	pub fn r(&self) -> &[u8] {
+		&self.0[0..32]
+	}
+
```

Comment:
Whitespace looks funky in this file (or is it just me), can you check it please?

Reply:
Double-checked and didn't find anything suspicious in the latest version of this file (there were spaces in the fist one)